gooddata-java/src/main/java/com/gooddata/sdk/service/hierarchicalconfig/HierarchicalConfigService.java

```diff
@@ -0,0 +1,121 @@
+
+            if (settings == null) {
+                throw new GoodDataException("empty response from API call");
+            }
+            return settings;
+        } catch (GoodDataException | RestClientException e) {
+            throw new GoodDataException("Unable to list project settings for project ID=" + project.getId(), e);
+        }
+    }
+
+    /**
+     * Returns setting for given project.
+     *
+     * @param project project, cannot be null
+     * @param settingName unique name (key) of setting item, cannot be empty
+     * @return setting for given project with given name (key)
+     */
+    public SettingItem getProjectSetting(final Project project, final String settingName) {
+        notNull(project, "project");
+        notEmpty(settingName, "settingName");
+
+        try {
+            String settingUri = PROJECT_SETTING_TEMPLATE.expand(project.getId(), settingName).toString();
+            SettingItem setting = restTemplate.getForObject(settingUri, SettingItem.class);
+
+            if (setting == null) {
+                throw new GoodDataException("Project setting cannot be retrieved from URI " + settingUri);
+            }
+            return setting;
+        } catch (GoodDataException | RestClientException e) {
+            throw new GoodDataException("Unable to get project setting: " + settingName, e);
+        }
+    }
+
+    /**
+     * Creates or update setting for given project.
+     *
+     * @param project project for which the setting item should be created/updated, cannot be null
+     * @param setting setting to be created/updated, cannot be null
+     * @return created/updated project setting
+     */
+    public SettingItem upsertProjectSetting(final Project project, final SettingItem setting) {
+        notNull(project, "project");
+        notNull(setting, "setting");
+
+        try {
+            // Validation of setting whether is it has a source which is allowed to be created via projects REST API
+            isItProjectSetting(setting);
+
+            String settingUri = PROJECT_SETTING_TEMPLATE.expand(project.getId(), setting.getName()).toString();
+            restTemplate.put(settingUri, setting);
+            SettingItem updated = restTemplate.getForObject(settingUri, SettingItem.class);
+
+            if (updated == null) {
+                throw new GoodDataException("Project setting cannot be retrieved from URI " + settingUri);
+            }
+            return setting;
+        } catch (GoodDataException | RestClientException e) {
+            throw new GoodDataException("Unable to create project setting: " + setting.getKey(), e);
+        }
+    }
+
+    /**
+     * Deletes existing project setting.
+     *
+     * @param setting existing project setting with links set properly, cannot be null
+     */
+    public void deleteProjectSetting(final SettingItem setting) {
```

Comment:
see naming in https://github.com/gooddata/gooddata-java/blob/master/CONTRIBUTING.md#services

Reply:
Fixed